core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -267,19 +270,34 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
     consumers.map(consumer => new ConsumerWrapper(consumer, customRebalanceListener, whitelist))
   }
 
-  def commitOffsets(consumerWrapper: ConsumerWrapper) {
+  def commitOffsets(consumerWrapper: ConsumerWrapper, retry: Int = Integer.MAX_VALUE): Unit = {
     if (!exitingOnSendFailure) {
       trace("Committing offsets.")
       try {
         consumerWrapper.commit()
+        lastSuccessfulCommitTime = time.milliseconds
       } catch {
         case e: WakeupException =>
           // we only call wakeup() once to close the consumer,
           // so if we catch it in commit we can safely retry
           // and re-throw to break the loop
-          consumerWrapper.commit()
+          commitOffsets(consumerWrapper, retry)
           throw e
 
+        case _: TimeoutException if retry > 0 =>
+          if (retry == Integer.MAX_VALUE) { // only try to remove offsets for nonexistent topics once
```

Comment:
What if topic deletion happens after a few retries? It seems safer to to filter topic for every retry. It is probably OK because the time used to filter topic in MM should be much smaller than the time needed for backoff and the time needed to wait for offset commit to pass, right?